server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileRepository.java

```diff
@@ -59,6 +60,14 @@ List<MemberProfile> search(@Nullable String firstName, @Nullable String middleNa
                                @Nullable String workEmail, @Nullable String supervisorId, @Nullable Boolean terminated);
 
     List<MemberProfile> findAll();
+    List<UUID> findSupervisoridByIdIn(Set<UUID> Ids);
+
+    @Query(value = """
+            SELECT PGP_SYM_DECRYPT(cast(workEmail as bytea), :aesKey) as workEmail
+                    FROM member_profile mp
+                    WHERE  mp.id IN (:ids)""",
+            nativeQuery = true)
+    List<String> findWorkEmailByIdIn(String aesKey, Set<String> ids);
```

Comment:
Can we drop the aesKey parameter from here, and switch to use property expansion as with the other queries

```suggestion
    @Query(value = """
            SELECT PGP_SYM_DECRYPT(cast(workEmail as bytea), '${aes.key}') as workEmail
                    FROM member_profile mp
                    WHERE  mp.id IN (:ids)""",
            nativeQuery = true)
    List<String> findWorkEmailByIdIn(Set<String> ids);
```

And then we don't need to inject the value in to `ReviewPeriodServicesImpl` on line 41, nor pass it as a parameter on line 145

Reply:
Unfortunately, there appears to be a bug in Micronaut Data when using the property expansion syntax with `where in` queries. The injected property gets included with the method arguments as a SQL argument, throwing off the number of arguments.

Reply:
Ahhhh! I see 👍